### ISSUE TITLE
[Question] Custom loss function

### ISSUE THREAD

**Issue by shawpan** [tensorflow/adanet#44]
how to use a custom loss function for regression? An example would be best. 
For example, loss function is 

``` \sum_{i=1}^{n} -log(\Phi(y_i - f(x_i))) ```

**Comment by cweill** [comment]
For your use case, I would recommend passing a custom `loss_fn` to [`tf.contrib.estimator.regression_head`](https://www.tensorflow.org/api_docs/python/tf/contrib/estimator/regression_head).

For example:

```python
def custom_loss(labels, logits, features):
  # The outputs may need to have the same shape as logits: [batch_size, logits_dimension].
  return -tf.log(phi(labels-logits))
  
# Don't forget to specify how to aggregate the losses across a batch with `loss_reduction`.
head = tf.contrib.estimator.regression_head(loss_fn=custom_loss)
```

**Comment by shawpan** [comment]
Thanks, Charles, I missed this parameter and `tf.contrib.learn.regression_head` is deprecated as I see. One more thing, can I assume to get a feature value from `features` variable like `features['featureName']` ? or does it have a different data structure?

**Comment by shawpan** [comment]
Ok, found it myself. Thanks for the help.
`features['featureName']` can be accessed but needed to reshape using `tf.reshape(features['featureName'], [-1, logits_dimension])`. Everything is a tensor so `1.0 + features['featureName']`  is also valid or use math functions `tf.math.add(1.0, features['featureName'])`
